### PR TITLE
Add missing dependencies, configure `maglnet/composer-require-checker`

### LINKED ISSUE #51: DriverFactory.php incompatible with doctrine/common: "^3.0"
Hi i noticed that /src/DriverFactory.php is not compatible with doctrine/common ^3.0
since 

use Doctrine\Common\Annotations\AnnotationReader;
use Doctrine\Common\Annotations\AnnotationRegistry;

are only available below version 3.0, so it seems the package.json is broken with psr-contrianer-doctrine 3.x

### PR DESCRIPTION
**Pull request by Gared**
* Add maglnet/composer-require-checker to GA testing
* add missing dependencies in require
  * doctrine/event-manager
  * psr/cache
  * doctrine/annotations (move from require-dev to require)

### CODE REVIEW

**Comment by Gared** [comment]
Fixes #51

**Comment by Gared** [comment]
Output of ComposerRequireChecker:
```
The following 7 unknown symbols were found:
+------------------------------------------------+--------------------+
| Unknown Symbol                                 | Guessed Dependency |
+------------------------------------------------+--------------------+
| Doctrine\Common\Annotations\AnnotationReader   |                    |
| Doctrine\Common\Annotations\AnnotationRegistry |                    |
| Doctrine\Common\Annotations\PsrCachedReader    |                    |
| Doctrine\Common\Annotations\Reader             |                    |
| Doctrine\Common\EventManager                   |                    |
| Doctrine\Common\EventSubscriber                |                    |
| Psr\Cache\CacheItemPoolInterface               |                    |
+------------------------------------------------+--------------------+
```

File: composer.json [hunk 1]
```diff
@@ -17,17 +17,20 @@
...
         "doctrine/migrations": "^3.4",
         "doctrine/orm": "^2.11",
         "doctrine/persistence": "^2.3",
+        "psr/cache": "^1.0",
         "psr/container": "^1.0 || ^2.0"
     },
     "require-dev": {
-        "doctrine/annotations": "^1.13",
         "doctrine/coding-standard": "^8.1",
+        "maglnet/composer-require-checker": "^3.8",
```
Comment (Ocramius) [inline composer.json]: This should really be part of our CI pipeline :D

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (Gared) [reply composer.json]: Do you want me to add this to your CI pipeline? I don't exactly know what do you want from me ^^

Comment (Ocramius) [reply composer.json]: Correct

Comment (Gared) [reply composer.json]: Done

**Review by Ocramius — APPROVED** [review]
Thanks @Gared!

**Pull request merged**